app/models/noise.rb

```diff
@@ -42,6 +40,9 @@ class Noise < ActiveRecord::Base
   PROVIDER_SIDEWALKS = 'sidewalks'
   PROVIDER_TWITTER = 'twitter'
 
+  delegate :name, :provider_url, to: :user, prefix: true, allow_nil: true
+  delegate :url_helpers, to: 'Rails.application.routes'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.